packages/core/diagnostic/src/diagnostic.js

```diff
@@ -63,7 +65,7 @@ export type Diagnostic = {|
   language?: string,
 
   /** A code frame points to a certain location(s) in the file this diagnostic is linked to (optional) */
-  codeFrame?: ?DiagnosticCodeFrame,
+  codeFrame?: ?DiagnosticCodeFrame | ?Array<DiagnosticCodeFrame>,
```

Comment:
We could possibly make this always an array if we wanted. I just didn't want to have to change all the tests. Any opinions?

Reply:
I think `BuildFailureEvent` originally also contained either a diagnostic or an array of them, and it was rather cumbersome to use because you always had to somehow normalize them into an array anyway yourself.

Reply:
Okay, I've refactored it to rename `codeFrame` to `codeFrames` and make it always an array. Also moved `language` and `filePath` into code frames rather than on the top-level diagnostic object.